app/controllers/products_controller.rb

```diff
@@ -0,0 +1,57 @@
+class ProductsController < ApplicationController
+
+  def index
+    if params[:price].present? || params[:quantity].present?
+      @products = Product.where('price = :price OR total_quantity = :quantity', price: params[:price].to_f,
+      quantity: params[:quantity].to_i)
+
+    elsif params[:price].present? && params[:quantity].present?
+      @products = Product.where('price = :price AND total_quantity = :quantity', price: params[:price].to_f,
+      quantity: params[:quantity].to_i)
+
+    else
+      @products = Product.all
+
+    end
+  end
+
+  def new
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.new(product_params)
+
+    if @product.save
+      redirect_to '/products'
+    else
+      render :new, status: :unprocessable_entity
+    end
```

Comment:
Let's do this as ternary condition as it's just one if else condition. 
Something like @product.save ? redirect_to '/products' : render :new, status: :unprocessable_entity